rich/_inspect.py

```diff
@@ -233,3 +233,29 @@ def _get_formatted_doc(self, object_: Any) -> Optional[str]:
         if not self.help:
             docs = _first_paragraph(docs)
         return escape_control_codes(docs)
+
+
+def object_types_tree(obj: Union[object, Type[Any]]) -> Tuple[type, ...]:
+    obj_type: Type[Any] = type(obj)
+    if obj_type is type:
+        # the given object was already a type: let's use this instead:
+        obj_type = cast(Type[Any], obj)  # I wish MyPy could infer this itself :-/
+    return getattr(obj_type, "__mro__", ())
+
+
+def object_types_tree_as_strings(obj: object) -> Tuple[str, ...]:
+    return tuple(
+        [
+            f'{getattr(type_, "__module__", "")}.{getattr(type_, "__qualname__", "")}'
+            for type_ in object_types_tree(obj)
+        ]
+    )
+
+
+def object_is_one_of_types(
```

Comment:
note sure about that function's name - but I couldn't find better for the moment :sweat_smile: